config/producer.properties

```diff
@@ -18,36 +18,28 @@
 
 # list of brokers used for bootstrapping knowledge about the rest of the cluster
 # format: host1:port1,host2:port2 ...
-metadata.broker.list=localhost:9092
+bootstrap.servers=localhost:9092
+
+# specify the compression codec for all data generated: none, gzip, snappy, lz4
+compression.type=none
 
 # name of the partitioner class for partitioning events; default partition spreads data randomly
 #partitioner.class=
 
-# specifies whether the messages are sent asynchronously (async) or synchronously (sync)
-producer.type=sync
-
-# specify the compression codec for all data generated: none, gzip, snappy, lz4.
-# the old config values work as well: 0, 1, 2, 3 for none, gzip, snappy, lz4, respectively
-compression.codec=none
-
-# message encoder
-serializer.class=kafka.serializer.DefaultEncoder
+# the maximum amount of time the client will wait for the response of a request
+#request.timeout.ms=
 
-# allow topic level compression
-#compressed.topics=
+# how long `KafkaProducer.send` and `KafkaProducer.partitionsFor` will block for
+#max.block.ms=
 
-############################# Async Producer #############################
-# maximum time, in milliseconds, for buffering data on the producer queue 
-#queue.buffering.max.ms=
+# the producer will wait for up to the given delay to allow other records to be sent so that the sends can be batched together
+#linger.ms=
 
-# the maximum size of the blocking queue for buffering on the producer 
-#queue.buffering.max.messages=
+# the maximum size of a request in bytes
+#max.request.size=
 
-# Timeout for event enqueue:
-# 0: events will be enqueued immediately or dropped if the queue is full
-# -ve: enqueue will block indefinitely if the queue is full
-# +ve: enqueue will block up to this many milliseconds if the queue is full
-#queue.enqueue.timeout.ms=
+# the default batch size in bytes when batching multiple records sent to a partition
+#batch.size=
```

Comment:
Maybe add linger.ms too? There are usually tuned together... Although TBH, I'm not sure where / why this example file will be used...